### PR TITLE
Feature/issue 1 domain class

### PR DESCRIPTION
**Pull request by bonotake**
According to #1, this PR adds a core class `Relation` and relevant classes.

### CODE REVIEW

File: relations/domain.py [hunk 1]
```diff
@@ -1,3 +1,43 @@
...
+    """The core domain class of relations."""
+
+    identifier: RelationID
+    relation_type: RelationType
+    e_print: EPrint
+    resource: Resource
+    description: str
+    added_at: datetime
+    creator: Optional[str]
+    superceded_or_suppressed: Optional[RelationID]
```
Comment (erickpeirson) [inline relations/domain.py]: Is this a reference to the a relation that is superceded or suppressed by this relation? Maybe put it in present tense, e.g. ``supercedes_or_suppresses``?

File: relations/domain.py [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+    arxiv_id: ArXivID
+    version: int
+
+
+class RelationType(Enum):
+    """Type of relations."""
+
+    ADD = auto()
+    EDIT = auto()
+    DEACTIVATE = auto()
```
Comment (erickpeirson) [inline relations/domain.py]: Should we call this ``SUPPRESS`` for consistency with L43?

**Review by erickpeirson — COMMENTED** [review]
This looks like an excellent starting point. Just a couple of minor comments on naming

File: relations/domain.py (hunk 1 above)
Comment (bonotake) [reply relations/domain.py]: 6d2a39de7dbea994c3f9f515bb57feca3a109ba7

File: relations/domain.py (hunk 2 above)
Comment (bonotake) [reply relations/domain.py]: 6d2a39de7dbea994c3f9f515bb57feca3a109ba7

**Review by erickpeirson — APPROVED** [review]
Looks good!

**Comment by erickpeirson** [comment]
Fixes #1

**Pull request merged**